extensions/pkg/controller/backupbucket/reconciler.go

```diff
@@ -90,6 +90,16 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	if bb.DeletionTimestamp != nil {
 		return r.delete(r.ctx, bb)
 	}
+
+	shoot, err := extensionscontroller.GetShoot(r.ctx, r.client, request.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if extensionscontroller.IsShootFailed(shoot) {
+		return reconcile.Result{}, nil
+	}
```

Comment:
Shouldn't this be moved up to line 89?

Reply:
I though:
1) Checking for a deletion timestamp on the CRD is cheaper than retrieving & parsing the Cluster CRD ->  in case the Extension CRD should be deleted, not necessary to do the following operation 
2) Ability to delete CRDs of failed clusters manually without changing the Cluster CRD first.

What would be the reason for moving it before the deletion check?

Reply:
I don't understand. The motivation behind the `Failed` status is that nothing is happening (not even deletion) unless the user manually confirms that it can be retried. 
For example, consider a shoot that is stuck in deletion for 24h because of invalid credentials or whatever other reason. Gardenlet would transition the status to `Failed` and would not try to delete it again until the user acts. The same behaviour should be implemented by the extensions.